catalog/core/catalog-core-federationstrategy/src/test/java/ddf/catalog/federation/impl/FederationStrategyTest.java

```diff
@@ -85,21 +89,54 @@
 
     private static final long LONG_TIMEOUT = 100;
 
-    private static final FilterFactory FILTER_FACTORY = new FilterFactoryImpl();
-
-    private static final ExecutorService EXECUTOR = Executors.newFixedThreadPool(2);
-
     private static final Logger LOGGER =
             LoggerFactory.getLogger(FederationStrategyTest.class.getName());
 
+    private FilterFactory filterFactory;
+
+    private ExecutorService executor;
+
     @Rule
     public PowerMockRule rule = new PowerMockRule();
 
+    @Before
+    public void setup() throws Exception {
+        filterFactory = new FilterFactoryImpl();
+        refreshExecutor();
+    }
+
+    @After
+    public void finish() throws Exception {
+        killAndWaitForExecutor();
+    }
+
+    /**
+     * Note that we ALWAYS call this upon terminating a test.
+     * @throws Exception
+     */
+    private void killAndWaitForExecutor() throws Exception {
+        if (executor != null && !executor.isShutdown()) {
+            executor.shutdown();
```

Comment:
I looked into it and it'll be risky. Assertions may fail. Let's air on the side of caution and keep it the way it is.